packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -730,3 +715,738 @@ class _RenderChildOverflowBox extends RenderAligningShiftedBox {
+    }
+  }
+
+  @override Color? get foregroundColor => _resolveColor(context, _FabPrimaryM3.iconColor);
+  @override Color? get backgroundColor => _resolveColor(context, _FabPrimaryM3.containerColor);
+  @override Color? get splashColor => _resolveColor(context, _FabPrimaryM3.pressedStateLayerColor, _FabPrimaryM3.pressedStateLayerOpacity);
+  @override double get elevation => _FabPrimaryM3.containerElevation;
+  @override Color? get focusColor => _resolveColor(context, _FabPrimaryM3.focusStateLayerColor, _FabPrimaryM3.focusStateLayerOpacity);
+  @override double get focusElevation => _FabPrimaryM3.focusContainerElevation;
+  @override Color? get hoverColor => _resolveColor(context, _FabPrimaryM3.hoverStateLayerColor, _FabPrimaryM3.hoverStateLayerOpacity);
+  @override double get hoverElevation => _FabPrimaryM3.hoverContainerElevation;
+  @override double get highlightElevation => _FabPrimaryM3.pressedContainerElevation;
+
+  @override
+  ShapeBorder? get shape => RoundedRectangleBorder(
+    borderRadius: BorderRadius.all(Radius.circular(borderRadius()))
+  );
+
+  @override bool? get enableFeedback => true;
+
+  @override
+  double? get iconSize {
+    switch (type) {
+      case _FloatingActionButtonType.regular: return _FabPrimaryM3.iconSize;
+      case _FloatingActionButtonType.small: return _FabSmallPrimaryM3.iconSize;
+      case _FloatingActionButtonType.large: return _FabLargePrimaryM3.iconSize;
+      case _FloatingActionButtonType.extended: return _FabExtendedPrimaryM3.iconSize;
+    }
+  }
+
+  @override
+  BoxConstraints? get sizeConstraints => const BoxConstraints.tightFor(
+    width: _FabPrimaryM3.containerWidth,
+    height: _FabPrimaryM3.containerHeight,
+  );
+
+  @override
+  BoxConstraints? get smallSizeConstraints => const BoxConstraints.tightFor(
+    width: _FabSmallPrimaryM3.containerWidth,
+    height: _FabSmallPrimaryM3.containerHeight,
+  );
+
+  @override
+  BoxConstraints? get largeSizeConstraints => const BoxConstraints.tightFor(
+    width: _FabLargePrimaryM3.containerWidth,
+    height: _FabLargePrimaryM3.containerHeight,
+  );
+
+  @override
+  BoxConstraints? get extendedSizeConstraints => const BoxConstraints.tightFor(
+    height: _FabSmallPrimaryM3.containerHeight,
+  );
+
+  @override double? get extendedIconLabelSpacing => 8.0;
+  @override EdgeInsetsGeometry? get extendedPadding => EdgeInsetsDirectional.only(start: hasChild && isExtended ? 16.0 : 20.0, end: 20.0);
+  @override TextStyle? get extendedTextStyle => _textStyleForValue(_FabExtendedPrimaryM3.labelTextFont as _TokenTypeScaleValue);
+}
+
+
+// ----------------------------------------------------------------------
+// Generated token data. In the case of the FAB, there are four variants,
+// with a lot of overlap between the values.
+//
+
+// GENERATED CODE - DO NOT MODIFY BY HAND
+// TO MODIFY THIS FILE UPDATE:
+//   mobile/flutter/tools/token_gen/script/lib/template/component.tmpl
+//
+// Version: v0.65
+
+class _FabSmallPrimaryM3 {
```

Comment:
Maybe we should always prefix these classes with `_Token` to make it more explicit (like the elevation and state ones below)